lib/core/db/PostgreSQLQueries.js

```diff
@@ -0,0 +1,146 @@
+        + "     usages smallint, "
+        + "     PRIMARY KEY (guild_id, user_id, sent_at, emote_id) "
+        + " ); ",
+    " CREATE TABLE IF NOT EXISTS custom_emote_metadata "
+        + " ( "
+        + "     id bigint PRIMARY KEY, "
+        + "     name text, "
+        + "     animated boolean, "
+        + "     guild_id bigint "
+        + " ); ",
+    " CREATE TABLE IF NOT EXISTS guilds "
+        + " ( "
+        + "     id bigint PRIMARY KEY "
+        + " ); ",
+    " CREATE TABLE IF NOT EXISTS channels "
+        + " ( "
+        + "     channel_id bigint PRIMARY KEY, "
+        + "     latest_parsed_id bigint, "
+        + "     earliest_parsed_id bigint, "
+        + "     latest_unparsed_id bigint "
+        + " ); ",
+];
+queries.emoteUsagesColumn = "usages";
+queries.customEmoteIdColumn = "id";
+queries.customEmoteNameColumn = "name";
+queries.customEmoteServerColumn = "guild_id";
+queries.channelsIdColumn = "channel_id";
+queries.wipeData = [
+    " TRUNCATE emotes; ",
+    " TRUNCATE custom_emote_metadata; ",
+    " TRUNCATE guilds; ",
+    " TRUNCATE channels; ",
+];
+queries.dropTables = [
+    " DROP TABLE IF EXISTS emotes; ",
+    " DROP TABLE IF EXISTS custom_emote_metadata; ",
+    " DROP TABLE IF EXISTS guilds; ",
+    " DROP TABLE IF EXISTS channels; ",
+];
+queries.createIndexes = [
+];
+queries.buildUpdateChannel = (channel_id, { latestParsed_id, earliestParsed_id, latestUnparsed_id }) => {
+    const params = [ channel_id ];
+    const columns = [ "channel_id" ];
+    const paramsQueries = [ "$1" ];
+    const columnUpdates = [];
+    if (latestParsed_id !== undefined) {
+        columns.push("latest_parsed_id");
+        params.push(latestParsed_id);
+        paramsQueries.push("$" + params.length);
+    }
+    if (earliestParsed_id !== undefined) {
+        columns.push("earliest_parsed_id");
+        params.push(earliestParsed_id);
+        paramsQueries.push("$" + params.length);
+    }
+    if (latestUnparsed_id !== undefined) {
+        columns.push("latest_unparsed_id");
+        params.push(latestUnparsed_id);
+        paramsQueries.push("$" + params.length);
+    }
+
+    for (const column of columns) {
+        columnUpdates.push(`${column} = EXCLUDED.${column}`);
+    }
+    const query = ` INSERT INTO channels (${columns.join(", ")}) VALUES (${paramsQueries.join(", ")}) `
+        + ` ON CONFLICT (channel_id) DO UPDATE SET ${columnUpdates.join(", ")};`;
+    
+    return { query, params };
+};
+queries.buildFetchChannels = (channel_ids) => {
+    const filters = channel_ids.length > 0 ? " WHERE channel_id = ANY ($1); " : " ; ";
+    const query = " SELECT "
+        + " channel_id, latest_parsed_id, earliest_parsed_id, latest_unparsed_id "
+        + " FROM channels " + filters;
+    const params = [ channel_ids ];
+
+    return { query, params };
+};
+queries.insertGuild = " INSERT INTO guilds (id) VALUES ($1) ON CONFLICT DO NOTHING; ";
+queries.fetchGuilds = " SELECT id FROM guilds; ";
```

Comment:
`fetchGuilds` seems to imply you're getting all of the guild metadata, not just IDs.

Reply:
There's only the ID in the table right now. If we do have other stuff later, we'll fetch everything.

Reply:
At that point, why not make it `SELECT * FROM guilds;`, so you don't have to keep editing it?

Reply:
(although we'd only use the ID where that query is used right now)

Reply:
I'd prefer for it to be renamed or for the query to be adjusted, but I'll leave it up to you.

Reply:
Because I prefer explicitely specifying which columns I'm fetching even if it's at the cost of editing the query everytime a column is added.

I prefer clarity the hundreds of times I'm gonna read that line at the cost of renaming it once or twice when columns are added.